neo/Plugins/Plugin.cs

```diff
@@ -39,23 +44,60 @@ public static bool CheckPolicy(Transaction tx)
 
         internal static void LoadPlugins(NeoSystem system)
         {
+            if (_configWatcher != null)
+            {
+                _configWatcher?.Dispose();
+                _configWatcher = null;
+            }
+
             System = system;
             string path = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), "Plugins");
             if (!Directory.Exists(path)) return;
+
+            _configWatcher = new FileSystemWatcher(path, "*.json")
+            {
+                EnableRaisingEvents = true,
+                IncludeSubdirectories = true,
+                NotifyFilter = NotifyFilters.CreationTime | NotifyFilters.LastWrite | NotifyFilters.Size,
+            };
+
+            _configWatcher.Changed += _configWatcher_Changed;
+            _configWatcher.Created += _configWatcher_Changed;
+
             foreach (string filename in Directory.EnumerateFiles(path, "*.dll", SearchOption.TopDirectoryOnly))
             {
                 Assembly assembly = Assembly.LoadFile(filename);
                 foreach (Type type in assembly.ExportedTypes)
                 {
                     if (!type.IsSubclassOf(typeof(Plugin))) continue;
                     if (type.IsAbstract) continue;
+
                     ConstructorInfo constructor = type.GetConstructor(Type.EmptyTypes);
                     if (constructor == null) continue;
-                    constructor.Invoke(null);
+
+                    var plugin = (Plugin)constructor.Invoke(null);
+                    plugin?.Configure(assembly.GetConfiguration());
                 }
             }
         }
 
+        private static void _configWatcher_Changed(object sender, FileSystemEventArgs e)
+        {
+            foreach (var plugin in Plugins)
+            {
+                if (plugin.ConfigFile == e.FullPath)
+                {
+                    plugin.Configure(plugin.GetType().Assembly.GetConfiguration());
+                    break;
+                }
+            }
+        }
+
+        public virtual void Configure(IConfigurationSection config)
```

Comment:
`abstract`?

Reply:
My first idea was abstract, but then ... you require to recompile all plugins, but i think that is not a problem, because the logic change